### PR TITLE
Add constraint from error function option to the trajopt default config

### PR DESCRIPTION
**Pull request by Levi-Armstrong**

### CODE REVIEW

File: tesseract/tesseract_planning/tesseract_motion_planners/src/trajopt/config/trajopt_planner_default_config.cpp [hunk 1]
```diff
@@ -187,6 +187,39 @@ bool TrajOptPlannerDefaultConfig::generate()
...
+                            "ConstraintErrFunc_" + std::to_string(i)));
+      }
+      else
+      {
+        prob->addConstraint(std::make_shared<trajopt::TrajOptConstraintFromErrFunc>(sco::VectorOfVector::construct(c.first),
+                            sco::MatrixOfVector::construct(c.second),
+                            prob->GetVarRow(s, 0, pci->kin->numJoints()),
+                            Eigen::VectorXd::Ones(0),
+                            sco::EQ,
+                            "ConstraintErrFunc_" + std::to_string(i)));
```
Comment (mpowelson) [inline tesseract/tesseract_planning/tesseract_motion_planners/src/trajopt/config/trajopt_planner_default_config.cpp]: I realize this is what I had, but I think this should be INEQ if we want the constraint to be anything > 0 to be error. Personnally I think I'd be inclined to keep this as EQ and just change the comment in the header

File: tesseract/tesseract_planning/tesseract_motion_planners/include/tesseract_motion_planners/trajopt/config/trajopt_planner_default_config.h [hunk 2]
```diff
@@ -95,6 +95,22 @@ struct TrajOptPlannerDefaultConfig : public TrajOptPlannerConfig
...
   bool smooth_jerks = true;
+
+  /** @brief Error function that is set as a constraint for each timestep.
+   *
+   * This is a vector of std::pair<Error Function, Error Function Jacobian>, the error function is required, but the
+   * jacobian is optional (nullptr).
+   *
+   * Error Function:
+   *   arg: VectorXd will be all of the joint values for one timestep.
+   *   return: VectorXd of violations for each joint. Anything > 0 will be a violation
```
Comment (mpowelson) [inline tesseract/tesseract_planning/tesseract_motion_planners/include/tesseract_motion_planners/trajopt/config/trajopt_planner_default_config.h]: See other comment

Comment (Levi-Armstrong) [reply tesseract/tesseract_planning/tesseract_motion_planners/include/tesseract_motion_planners/trajopt/config/trajopt_planner_default_config.h]: I changed it to anything != 0 will be a violation.

**Comment by Levi-Armstrong** [comment]
@mpowelson You good with the changes?

**Pull request merged**